sdk/servicebus/azure-servicebus/azure/servicebus/_common/receiver_mixins.py

```diff
@@ -176,3 +178,11 @@ def _on_attach(self, source, target, properties, error):
     def _populate_message_properties(self, message):
         if self._session:
             message[MGMT_REQUEST_SESSION_ID] = self._session_id
+
+    def _enhanced_message_received(self, message):
```

Comment:
We should take notes for how we can make this kind of fix easier in the new pyamqp....
For example have the `on_message_received` callback be more easily replaceable without having to set all these private internal attributes......

Reply:
totally agreed, will add notes!